src/reuse/modules/mobile/device.ts

```diff
@@ -0,0 +1,116 @@
+   */
+  async isAppInstalled(packageIdorBundleId: string): Promise<boolean> {
+    const vl = this.vlf.initLog(this.isAppInstalled);
+    try {
+      const isAppInstalledinDevice: boolean = browser.isAppInstalled(packageIdorBundleId);
+      vl.log(`Given app package/bundle id ${packageIdorBundleId} installed on the device is ${isAppInstalledinDevice.toString()}`);
+      return isAppInstalledinDevice;
+    } catch (error) {
+      return this.ErrorHandler.logException(error, 'Error: During isAppInstalled', true);
+    }
+  }
+
+/**
+   * @function installAppForPlatform
+   * @memberOf mobile.device
+   * @description Install the appropriate app based on the platform the test is being executed on.
+   * @param {string} appPath - Path of the app(.apk, .ipa)
+   * @example 
+   * await mobile.device.installAppForPlatform("/path/to/your/app.apk");
+   * await mobile.device.installAppForPlatform("/path/to/your/app.ipa");
+   */
+async installAppForPlatform(appPath: string): Promise<void> {
+  const vl = this.vlf.initLog(this.installAppForPlatform);
+  const platform = await browser.capabilities.platformName;
+  try {
+      if (platform.toLowerCase() === 'android') {
+          vl.log("Installing Android app...");
+          await browser.installApp(appPath);
+          vl.log("Android app installed successfully.");
+      } else if (platform.toLowerCase() === 'ios') {
+          vl.log("Installing iOS app...");
+          await browser.installApp(appPath);
+          vl.log("iOS app installed successfully.");
+      } else {
+          throw new Error(`Unsupported platform while installing the app: ${platform}`);
+      }
+  } catch (error) {
+      this.ErrorHandler.logException(error, 'Error: During installAppForPlatform', true);
+  }
+}
+
+/**
+   * @function pressKeyByNameInAndroid
+   * @memberOf mobile.device
+   * @description Simulate pressing a hardware key on the device (e.g., back button, home button, etc.), 
+   * and it specific to Android devices
+   * @param {string} keyName - The name of the key (e.g., "back", "home", "volumeUp", etc.)
+   * @example 
+   * await mobile.device.pressKeyByNameInAndroid("back");
+   * await mobile.device.pressKeyByNameInAndroid("home");
+   * await mobile.device.pressKeyByNameInAndroid("volumeUp");
+   * await mobile.device.pressKeyByNameInAndroid("volumeDown");
+   * await mobile.device.pressKeyByNameInAndroid("power");
+   * await mobile.device.pressKeyByNameInAndroid("enter");
+   * await mobile.device.pressKeyByNameInAndroid("space");
+   * await mobile.device.pressKeyByNameInAndroid("delete");
+   * await mobile.device.pressKeyByNameInAndroid("menu");
+   * await mobile.device.pressKeyByNameInAndroid("search");
+   * await mobile.device.pressKeyByNameInAndroid("camera");
+   * await mobile.device.pressKeyByNameInAndroid("focus");
+   * await mobile.device.pressKeyByNameInAndroid("notification");
+   */
+async pressKeyByNameInAndroid(keyName: string): Promise<void> {
```

Comment:
One thought to consider: maybe it would be better to organize the whole mobile namespace in a way that we have a sub namespace for each mobile platform? Then it would be `mobile.android.pressKeyByName` and `mobile.ios.pressKeyByName`. This would make sense to me, but I don't know all the nuances of mobile testing. @sirikumarsap @Templada what do you think?

Reply:
agree, everything which is specific for a certain platform should be moved to the corresponding (new) namespace -> mobile.ios.; mobile.android and so on. For functions which are platform independent or same we should have a module named common

@sirikumarsap

Reply:
interesting to know would be if the elements also need to be located differently from platform to platform. 

If so we would need another submodules as well under mobile.<platform>.element

Reply:
As per this review comments, these specific functions are now moved to device platform-wise, now mobile.android
mobile.ios 
files are created.